src/res_utf8.ml

```diff
@@ -24,118 +24,365 @@ let h4 = 0b1111_0000
 
 let cont_mask = 0b0011_1111
 
-type category = {
-  low: int;
-  high: int;
-  size: int;
-}
+type category = {low: int; high: int; size: int}
 
 let locb = 0b1000_0000
 let hicb = 0b1011_1111
 
-let categoryTable = [|
-(* 0 *) {low = -1; high= -1; size= 1}; (* invalid *)
-(* 1 *) {low = 1; high= -1; size= 1}; (* ascii *)
-(* 2 *) {low = locb; high= hicb; size= 2};
-(* 3 *) {low = 0xA0; high= hicb; size= 3};
-(* 4 *) {low = locb; high= hicb; size= 3};
-(* 5 *) {low = locb; high= 0x9F; size= 3};
-(* 6 *) {low = 0x90; high= hicb;  size= 4};
-(* 7 *) {low = locb; high= hicb; size= 4};
-(* 8 *) {low = locb; high= 0x8F; size= 4};
+let categoryTable =
+  [|
+    (* 0 *) {low = -1; high = -1; size = 1};
+    (* invalid *)
+    (* 1 *)
+    {low = 1; high = -1; size = 1};
+    (* ascii *)
+    (* 2 *)
+    {low = locb; high = hicb; size = 2};
+    (* 3 *) {low = 0xA0; high = hicb; size = 3};
+    (* 4 *) {low = locb; high = hicb; size = 3};
+    (* 5 *) {low = locb; high = 0x9F; size = 3};
+    (* 6 *) {low = 0x90; high = hicb; size = 4};
+    (* 7 *) {low = locb; high = hicb; size = 4};
+    (* 8 *) {low = locb; high = 0x8F; size = 4};
+  |]
 
-|]
-
-let categories = [|
-  1; 1; 1; 1; 1; 1 ;1 ;1 ;1 ;1 ;1 ;1 ;1 ;1 ;1 ;1;
-  1; 1; 1; 1; 1; 1 ;1 ;1 ;1 ;1 ;1 ;1 ;1 ;1 ;1 ;1;
-  1; 1; 1; 1; 1; 1 ;1 ;1 ;1 ;1 ;1 ;1 ;1 ;1 ;1 ;1;
-  1; 1; 1; 1; 1; 1 ;1 ;1 ;1 ;1 ;1 ;1 ;1 ;1 ;1 ;1;
-  1; 1; 1; 1; 1; 1 ;1 ;1 ;1 ;1 ;1 ;1 ;1 ;1 ;1 ;1;
-  1; 1; 1; 1; 1; 1 ;1 ;1 ;1 ;1 ;1 ;1 ;1 ;1 ;1 ;1;
-  1; 1; 1; 1; 1; 1 ;1 ;1 ;1 ;1 ;1 ;1 ;1 ;1 ;1 ;1;
-  1; 1; 1; 1; 1; 1 ;1 ;1 ;1 ;1 ;1 ;1 ;1 ;1 ;1 ;1;
-
-  0; 0; 0; 0;0; 0; 0; 0;0; 0; 0; 0;0; 0; 0; 0;
-  0; 0; 0; 0;0; 0; 0; 0;0; 0; 0; 0;0; 0; 0; 0;
-  0; 0; 0; 0;0; 0; 0; 0;0; 0; 0; 0;0; 0; 0; 0;
-  0; 0; 0; 0;0; 0; 0; 0;0; 0; 0; 0;0; 0; 0; 0;
-  (* surrogate range U+D800 - U+DFFFF = 55296 - 917503 *)
-  0; 0; 2; 2;2; 2; 2; 2;2; 2; 2; 2;2; 2; 2; 2;
-   2; 2; 2; 2; 2; 2; 2; 2; 2; 2; 2; 2; 2; 2; 2; 2;
-   3; 4; 4; 4; 4; 4; 4; 4; 4; 4; 4; 4; 4; 5; 4; 4;
-  6; 7; 7 ;7; 8; 0; 0; 0; 0; 0; 0; 0; 0; 0; 0; 0;
-|]
+let categories =
+  [|
+    1;
+    1;
+    1;
```

Comment:
@cristianoc This is not ideal. Probably should put the tables into a separate file that is excluded from formatting.

Reply:
Ah there is an easier way with `[@@ocamlformat "disable"]`.